zanata-common-api/src/main/java/org/zanata/rest/dto/stats/contribution/BaseContributionStatistic.java

```diff
@@ -10,25 +13,27 @@
  * @author Alex Eng <a href="mailto:aeng@redhat.com">aeng@redhat.com</a>
  */
 @JsonIgnoreProperties(ignoreUnknown = true)
+@JsonSerialize(include = JsonSerialize.Inclusion.NON_NULL)
+@JsonPropertyOrder({ "approved", "rejected", "translated", "needReview" })
 public class BaseContributionStatistic implements Serializable {
 
-    private int approved;
-    private int needReview;
-    private int translated;
-    private int rejected;
+    private Integer approved;
+    private Integer needReview;
+    private Integer translated;
+    private Integer rejected;
```

Comment:
Maybe add `@Nullable` too, just to be explicit (assuming that's why we are
using Integer).

On 11 January 2016 at 09:07, Carlos Munoz <notifications@github.com> wrote:

> In
> zanata-common-api/src/main/java/org/zanata/rest/dto/stats/contribution/BaseContributionStatistic.java
> <https://github.com/zanata/zanata-api/pull/40#discussion_r49283267>:
>
> >  public class BaseContributionStatistic implements Serializable {
> >
> > -    private int approved;
> > -    private int needReview;
> > -    private int translated;
> > -    private int rejected;
> > +    private Integer approved;
> > +    private Integer needReview;
> > +    private Integer translated;
> > +    private Integer rejected;
>
> Lets add a comment about why these are Integers and not ints. Since the
> API is sound, we can change the implementation later if we want to.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-api/pull/40/files#r49283267>.
>



-- 
Sean Flanigan

Principal Software Engineer
Globalisation Tools Engineering
Red Hat